sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -134,15 +183,27 @@ public static void VerifyNoIssueReported(Compilation compilation, DiagnosticAnal
                 .Result;
 
             VerifyNoExceptionThrown(diagnostics);
-
+            if (checkMode == CheckMode.CheckCompilationErrors)
+            {
+                VerifyBuildErrors(diagnostics, compilation);
+            }
             return diagnostics;
         }
 
+        private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Compilation compilation)
+        {
+            var buildErrors = diagnostics.Where(d => d.Id.StartsWith("CS") && d.Severity == DiagnosticSeverity.Error);
+            var expectedBuildErrors = new IssueLocationCollector()
+                .GetExpectedBuildErrors(compilation.SyntaxTrees.Skip(1).First().GetText().Lines)
+                .ToList();
+            CompareActualToExpected(buildErrors, expectedBuildErrors, true);
+        }
+
         private static void VerifyNoExceptionThrown(IEnumerable<Diagnostic> diagnostics) =>
             diagnostics.Where(d => d.Id == AnalyzerFailedDiagnosticId).Should().BeEmpty();
 
         private static void VerifyIssue(IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
-            Location location, string message, out string issueId)
+            Location location, string message, string extraInfo, out string issueId)
```

Comment:
Why not create a separate method `VerifyCompilerError`? Only half of `VerifyIssue` is used when verifying errors and I think we don't need it to be more complex than it is now.